skimage/morphology/selem.py

```diff
@@ -29,31 +29,67 @@ def square(width, dtype=np.uint8):
     return np.ones((width, width), dtype=dtype)
 
 
-def rectangle(nrows, ncols, dtype=np.uint8):
+def rectangle(nrows=None, ncols=None, dtype=np.uint8, width=None, height=None):
     """Generates a flat, rectangular-shaped structuring element.
 
     Every pixel in the rectangle generated for a given width and given height
     belongs to the neighborhood.
 
     Parameters
     ----------
-    height : int
-        The height of the rectangle.
-    width : int
-        The width of the rectangle.
+    nrows : int
+        The number of rows of the rectangle.
+    ncols : int
+        The number of columns of the rectangle.
 
     Other Parameters
     ----------------
     dtype : data-type
         The data type of the structuring element.
+    width : int, depreciated
+        The number of rows of the rectangle.
+    height : int, depreciated
+        The number of columns of the rectangle.
 
     Returns
     -------
     selem : ndarray
         A structuring element consisting only of ones, i.e. every
         pixel belongs to the neighborhood.
 
+    Notes
+    -----
+    ``nrows`` and ``ncols`` have None specified as a default value. This exists
+    for backwards compatibility with scikit-image v0.18 and older. You have to
+    specify values for both variables, and an exception will be raised
+    otherwise.
+
     """
+
+    if width is not None:
+        nrows = width
+
+        import warnings
+        warnings.warn("The usage of width is depreciated and the argument will"
+                      " be removed in a future version. Use"
+                      " `rectangle(nrows, ncols)` instead.",
+                      DeprecationWarning)
```

Comment:
We use `FutureWarning` by default because DeprecationWarnings are usually not displayed by Python.